### DISCUSSION TITLE
Do not convergence of heat conduction

### DISCUSSION THREAD

**Discussion by Salma-Mao** [idaholab/moose#20891] [Q&A General] [answered]
Hi,
I am studying  about heat conduction. Material properties change with temperature as shown in the picture.  I define the material like:

```
if （temperature[_qp] < 1000）
{
     specific_heat[_qp] = 1.23*temperature[_qp] + 1.23;
}
else
{
     specific_heat[_qp] = 4.56*temperature[_qp] + 4.56;
}

```

However, does not converge after a certain time step. At the same time, some units are just above 1000 K.  I guess this error has something to do with my judgment on temperature. But I do not know how to  solve this problem. 

![e1d335e82fe33116b3beb81df93a38e](https://user-images.githubusercontent.com/54804946/165536396-34c28a7a-3c1c-4102-8520-df60c3de295e.png)

Thanks,
MS

**Comment by GiudGiud** [comment, marked as answer]
Hello

From what you copy pasted it looks like the code skips the center portion of this graph. It jumps straight from 1200 J/m3 to 4500.
Is this normal? It seems to miss the T=1200 to 3500 portion.

if the solve struggles with too much discontinuity, you could smooth this profile.

Guillaume

**Reply by Salma-Mao** [reply]
Thanks  for your help.
